clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -484,6 +501,9 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         }
 #endif
         shard->prev_instr_ = memref;
+        if (cur_instr_decoded != nullptr) {
+            shard->prev_instr_decoded_ = cur_instr_decoded.get();
```

Comment:
Unless you use a `std::move`, the object pointed by cur_instr_decoded will be de-allocated when cur_instr_decoded goes out of scope.

Reply:
Done.